src/lol-ranks.js

```diff
@@ -216,28 +243,75 @@ class LoLRanks {
-          formattedTier = soloQueueRankData.tier.charAt(0) + soloQueueRankData.tier.slice(1).toLowerCase();
+        let soloQueueData = '';
+        let checkValue = '';
+        let tierValue = { 'Unranked':'0', 'Iron':'1', 'Bronze':'2', 'Silver':'3', 'Gold':'4', 'Platinum':'5', 'Diamond':'6', 'Master':'7', 'Grandmaster':'8', 'Challenger':'9' };
+        let rankValue = { 'IV':'0', 'III':'1', 'II':'2', 'I':'4' };
+
+        if (tierData === 'Unranked') {
+          checkValue = tierValue[tierData];
+          soloQueueData = '';
+        } else {
+          tierData = soloQueueRankData.tier.charAt(0) + soloQueueRankData.tier.slice(1).toLowerCase();
+          soloQueueData = soloQueueRankData.rank;
+          checkValue = tierValue[tierData] + rankValue[soloQueueData];
         }
 
-        if (formattedTier) {
-          const role = message.guild.roles.cache.find((r) => r.name === formattedTier);
+        let tierLocale = i18n.__('ranks');
+        let translatedTier = tierLocale[tierData];
 
-          this.updatePlayer(discordID, { rank: formattedTier });
+        if (translatedTier) {
+          const role = message.guild.roles.cache.find((r) => r.name === translatedTier);
+          const TierIcon = this.client.emojis.cache.find(emoji => emoji.name === tierData);
+          // Compare checkValue with player.totalValue
+          if (this.config.enableTierIconsOnMessages) {
+            if (player.totalValue !== null && player.totalValue > checkValue) {
+              await roleChannel.send(`${member.user}` + i18n.__('levelDown') + `${TierIcon} **${translatedTier} ${soloQueueData}**!`);
+            }
+
+            if (player.totalValue !== null && player.totalValue < checkValue) {
+              await roleChannel.send(`${member.user}` + i18n.__('levelUp') + `${TierIcon} **${translatedTier} ${soloQueueData}**!`);
+            }
+
+          } else {
+
+            if (player.totalValue !== null && player.totalValue > checkValue) {
+              await roleChannel.send(`${member.user}` + i18n.__('levelDown') + `**${translatedTier} ${soloQueueData}**!`);
+            }
+
+            if (player.totalValue !== null && player.totalValue < checkValue) {
+              await roleChannel.send(`${member.user}` + i18n.__('levelUp') + `**${translatedTier} ${soloQueueData}**!`);
+            }
+          }
+
+
+          // Updates player tier, rank and totalValue
+          this.updatePlayer(discordID, { tier: translatedTier, rank: soloQueueData, totalValue: checkValue });
           player = this.getPlayer(discordID);
 
           if (member.roles.cache.find(r => r.id === role.id)) {
-            reply += i18n.__('reply4_1') + `${formattedTier} ${soloQueueRankData ? soloQueueRankData.rank : ''}` + i18n.__('reply4_2');
+            if (this.config.enableTierIconsOnMessages) {
+              reply += i18n.__('reply4_1') + `${TierIcon} ${translatedTier}` + ` ${soloQueueRankData ? soloQueueData : ''} ` + i18n.__('reply4_2');
+            } else {
```

Comment:
It makes no sense to use string concatation together with string interpolation and please try to get rid of the duplicated code ("redundant code") by the nullish coalescing stuff I explained to you.